ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSfilemanip.cpp

```diff
@@ -162,10 +164,11 @@ int file_copy(std::string fname, std::string newname) {
 }
 
 int directory_exists(std::string dname) {
-  DWORD dwAttrib = GetFileAttributes(dname.c_str());
-
-  return (dwAttrib != INVALID_FILE_ATTRIBUTES &&
-         (dwAttrib & FILE_ATTRIBUTE_DIRECTORY));
+  DWORD file_attr;
+  tstring tstr_dname = widen(dname);
+  file_attr = GetFileAttributesW(tstr_dname.c_str());
+  return (file_attr != INVALID_FILE_ATTRIBUTES &&
+    (file_attr & FILE_ATTRIBUTE_DIRECTORY));
```

Comment:
You could have done this for `file_exists` too. §4.7/4 of the C++ standard states the following:
>If the source type is bool, the value **false** is converted to zero and the value **true** is converted to one.

Reply:
i decided not to do that because the genius who originally wrote the function didn't know this so to play it safe and reduce the chances of needing a review i kept that the same since it was previously merged.

Reply:
Not a huge deal, I was going to fix it recently myself but avoided it for the same reason you did. I just saw you changed the other one so may as well change this one too. It's a small thing, not a big deal, it's a correct change.

Reply:
We're more picky on pull requests larger than this one, like 500 or more lines. Or we're also pickier when it comes to changes being scattered around. In this case, more than half of this is diff is contained within a new extension you just created, so it's literally impossible you broke any _existing_ code. I hope that helps clarify that there is a reason for being pickier or less picky at certain times.